src/sqlancer/citus/CitusProvider.java

```diff
@@ -0,0 +1,444 @@
+    }
+
+    public enum Action implements AbstractAction<PostgresGlobalState> {
+        ANALYZE(PostgresAnalyzeGenerator::create), //
+        ALTER_TABLE(g -> CitusAlterTableGenerator.create(g.getSchema().getRandomTable(t -> !t.isView()), g,
+                generateOnlyKnown)), //
+        CLUSTER(PostgresClusterGenerator::create), //
+        COMMIT(g -> {
+            Query query;
+            if (Randomly.getBoolean()) {
+                query = new QueryAdapter("COMMIT", true);
+            } else if (Randomly.getBoolean()) {
+                query = PostgresTransactionGenerator.executeBegin();
+            } else {
+                query = new QueryAdapter("ROLLBACK", true);
+            }
+            return query;
+        }), //
+        CREATE_STATISTICS(PostgresStatisticsGenerator::insert), //
+        DROP_STATISTICS(PostgresStatisticsGenerator::remove), //
+        DELETE(CitusDeleteGenerator::create), //
+        DISCARD(PostgresDiscardGenerator::create), //
+        DROP_INDEX(PostgresDropIndexGenerator::create), //
+        INSERT(CitusInsertGenerator::insert), //
+        UPDATE(CitusUpdateGenerator::create), //
+        TRUNCATE(PostgresTruncateGenerator::create), //
+        VACUUM(PostgresVacuumGenerator::create), //
+        REINDEX(PostgresReindexGenerator::create), //
+        SET(CitusSetGenerator::create), //
+        CREATE_INDEX(CitusIndexGenerator::generate), //
+        SET_CONSTRAINTS((g) -> {
+            StringBuilder sb = new StringBuilder();
+            sb.append("SET CONSTRAINTS ALL ");
+            sb.append(Randomly.fromOptions("DEFERRED", "IMMEDIATE"));
+            return new QueryAdapter(sb.toString());
+        }), //
+        RESET_ROLE((g) -> new QueryAdapter("RESET ROLE")), //
+        COMMENT_ON(PostgresCommentGenerator::generate), //
+        RESET((g) -> new QueryAdapter("RESET ALL") /*
+                                                    * https://www.postgresql.org/docs/devel/sql-reset.html TODO: also
+                                                    * configuration parameter
+                                                    */), //
+        NOTIFY(PostgresNotifyGenerator::createNotify), //
+        LISTEN((g) -> PostgresNotifyGenerator.createListen()), //
+        UNLISTEN((g) -> PostgresNotifyGenerator.createUnlisten()), //
+        CREATE_SEQUENCE(PostgresSequenceGenerator::createSequence), //
+        CREATE_VIEW(CitusViewGenerator::create), //
+        QUERY_CATALOG((g) -> PostgresQueryCatalogGenerator.query());
+
+        private final QueryProvider<PostgresGlobalState> queryProvider;
+
+        Action(QueryProvider<PostgresGlobalState> queryProvider) {
+            this.queryProvider = queryProvider;
+        }
+
+        @Override
+        public Query getQuery(PostgresGlobalState state) throws SQLException {
+            return queryProvider.getQuery(state);
+        }
+    }
+
+    private static int mapActions(PostgresGlobalState globalState, Action a) {
```

Comment:
Since I cannot have the CitusProvider.Action enum extend PostgresProvider.Action, I cannot make a call to the function PostgresProvider.mapActions(), as it cannot cast CitusProvider.Action to PostgresProvider.Action